src/libsync/configfile.cpp

```diff
@@ -143,6 +145,86 @@ bool copy_dir_recursive(QString from_dir, QString to_dir)
     return true;
 }
 
+// Since QWindow has no saveGeometry, let's roll out our own,
+// this is obviously less smart since we can't query the layout system
+QByteArray saveWindowGeometry(QWindow *w)
+{
+    QByteArray array;
+    QDataStream stream(&array, QIODevice::WriteOnly);
+    stream.setVersion(QDataStream::Qt_5_12);
+    const quint32 magicNumber = 0x430A7465;
+    // Version history:
+    // - Nextcloud Client 3.1.0 : Version 1
+    const quint16 version = 1;
+    const int screenNumber = QGuiApplication::screens().indexOf(w->screen());
+    stream << magicNumber
+           << version
+           << w->geometry()
+           << qint32(screenNumber);
+    return array;
+}
+
+// This is almost a straight copy from Qt code
+static void checkRestoredGeometry(const QRect &availableGeometry, QRect *restoredGeometry)
+{
+    const auto frameHeight = 20;
+
+    if (!restoredGeometry->intersects(availableGeometry)) {
+        restoredGeometry->moveBottom(qMin(restoredGeometry->bottom(), availableGeometry.bottom()));
+        restoredGeometry->moveLeft(qMax(restoredGeometry->left(), availableGeometry.left()));
+        restoredGeometry->moveRight(qMin(restoredGeometry->right(), availableGeometry.right()));
+    }
+    restoredGeometry->moveTop(qMax(restoredGeometry->top(), availableGeometry.top() + frameHeight));
+}
+
+// Since QWindow has no restoreGeometry, let's roll out our own,
+// this is obviously less smart since we can't query the layout system
+bool restoreWindowGeometry(QWindow *w, const QByteArray &geometry)
+{
+    if (geometry.size() < 4)
+        return false;
+    QDataStream stream(geometry);
+    stream.setVersion(QDataStream::Qt_5_12);
+
+    const quint32 magicNumber = 0x430A7465;
```

Comment:
@er-vin Move this into static const since it is hardcoded in two methods?

Reply:
Anonymous namespace and constexpr but yes, good point. I'll add a commit for that.

Reply:
Done